ogzaf/macro_params.py

```diff
@@ -21,57 +23,136 @@ def get_macro_params():
+        #"10 year govt bond rate": "IRLTLT01ZAM156N",
+        "Total gov transfer payments": "B087RC1Q027SBEA",
+        "Social Security payments": "W823RC1",
         "Gov expenditures": "INDGFCEQDSMEI",
-        # "Gov interest payments": "A091RC1Q027SBEA",
+        "Gov interest payments": "A091RC1Q027SBEA",
         "Real GDP": "RGDPNAINA666NRUG",
         "Nominal GDP": "MKTGDPINA646NWDB",
     }
 
     # pull series of interest using pandas_datareader
-    fred_data = web.DataReader(variable_dict.values(), "fred", start, end)
+    fred_data = web.DataReader(fred_variable_dict.values(), "fred", start, end)
     fred_data.rename(
-        columns=dict((y, x) for x, y in variable_dict.items()), inplace=True
+        columns=dict((y, x) for x, y in fred_variable_dict.items()), inplace=True
     )
 
-    # make sure all dollar value data are in billions
-    fred_data["Debt held by public"] = fred_data["Debt held by public"] / 1000
-
-    # Separate quartely, monthly, and annual dataseries
+    # Separate quartely, monthly, and annual FRED dataseries
     fred_data_q = (
         fred_data[
             [
-                "Debt held by public",
                 "Nominal GDP",
                 "Total gov transfer payments",
                 "Social Security payments",
                 "Gov expenditures",
                 "Gov interest payments",
-                "GDP Per Capita",
             ]
         ]
         .resample("Q")
         .mean()
     )
-    fred_data_a = fred_data[["Labor share"]]
-    fred_data_d = fred_data[["BAA Corp Bond Rates", "10 year govt bond rate"]]
+
+    #fred_data_d = fred_data[["BAA Corp Bond Rates", "10 year govt bond rate"]]
 
     # initialize a dictionary of parameters
     macro_parameters = {}
 
     # print(fred_data.loc(str(baseline_date)))
     # find initial_debt_ratio
-    macro_parameters["initial_debt_ratio"] = pd.Series(
-        fred_data_q["Debt held by public"] / fred_data_q["Nominal GDP"]
-    ).loc[baseline_date]
+    macro_parameters["initial_debt_ratio"] = (
+        pd.Series(
+        wb_data_q["Gross PSD Gen Gov - percentage of GDP"]
+        ).loc[baseline_yearquarter]
+        )/100
+
+
+    # find initial_foreign_debt_ratio
+    macro_parameters["initial_foreign_debt_ratio"] = pd.Series(
+        wb_data_q["Gross PSD USD - external creditors"] / 
+        (
+            wb_data_q["Gross PSD USD - domestic creditors"]
+            + wb_data_q["Gross PSD USD - external creditors"]
+        )
+    ).loc[baseline_yearquarter] 
+
+    # find zeta_D (Share of new debt issues from government that are purchased by foreigners)
+    macro_parameters["zeta_D"] = pd.Series(
+        wb_data_q["Gross PSD USD - external creditors"] / 
+        (
+            wb_data_q["Gross PSD USD - domestic creditors"]
+            + wb_data_q["Gross PSD USD - external creditors"]
+        )
+    ).loc[baseline_yearquarter]
 
     # find alpha_T
     macro_parameters["alpha_T"] = pd.Series(
```

Comment:
The IMF Government Finance Statistics (GFS) has ZAF data on different items of government spending (overall expenditure, grants, subsidies, social security, social assistance, interest payments, etc) to calculate alpha_T and alpha_G. It used to have a SDMX API, but it no longer works. Now the data has to be retrieved manually from the GFS platform: https://data.imf.org/?sk=a0867067-d23c-4ebc-ad23-d3b015045405